packages/manager/modules/pci/src/projects/project/storages/containers/containers.controller.js

```diff
@@ -1,14 +1,44 @@
 export default class PciStoragesContainersController {
   /* @ngInject */
-  constructor($translate, CucCloudMessage) {
+  constructor($translate, $http, CucCloudMessage) {
     this.$translate = $translate;
+    this.$http = $http;
     this.CucCloudMessage = CucCloudMessage;
+    this.publicToggleLoading = false;
   }
 
   $onInit() {
     this.loadMessages();
   }
 
+  onPublicToggle(data) {
+    this.publicToggleLoading = true;
+    return this.$http
+      .put(`/cloud/project/${this.projectId}/storage/${data.id}`, {
+        containerType: data.public ? 'private' : 'public',
```

Comment:
do you mean containerType: data.public ? 'public' : 'private', ??

Reply:
If `public`, we send `private` and if `private` we change it to `public`. Its a toggle feature